enhancements/installer/aws-custom-region-and-endpoints.md

```diff
@@ -0,0 +1,321 @@
+
+  // Azure contains settings specific to the Azure infrastructure provider.
+  // +optional
+  Azure *AzurePlatformSpec `json:"azure,omitempty"`
+
+  // GCP contains settings specific to the Google Cloud Platform infrastructure provider.
+  // +optional
+  GCP *GCPPlatformSpec `json:"gcp,omitempty"`
+
+  // BareMetal contains settings specific to the BareMetal platform.
+  // +optional
+  BareMetal *BareMetalPlatformSpec `json:"baremetal,omitempty"`
+
+  // OpenStack contains settings specific to the OpenStack infrastructure provider.
+  // +optional
+  OpenStack *OpenStackPlatformSpec `json:"openstack,omitempty"`
+
+  // Ovirt contains settings specific to the oVirt infrastructure provider.
+  // +optional
+  Ovirt *OvirtPlatformSpec `json:"ovirt,omitempty"`
+}
+
+
+// AWSPlatformSpec holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformSpec struct {
+  // APIEndpoints list contains custom endpoints which will override default
+  // service endpoint of AWS Services.
+  // There must be only one CustomEndpoint for a service.
+  // +optional
+  APIEndpoints []AWSCustomEndpoint `json:"apiEndpoints,omitempty"`
+}
+
+// AWSCustomEndpoint store the configuration of a custom url to
+// override existing defaults of AWS Services.
+// Currently Supports - EC2, IAM, ELB, S3 and Route53.
+type AWSCustomEndpoint struct {
+    Service string `json:"service"`
+
+    // This must be a HTTPS URL
+    URL     string `json:"url"`
+}
+```
+
+##### Drawbacks / Limitations / Future work
+
+1. Since there is no controller in the cluster that manages the state of `Infrastructure` object, we cannot ensure that the API endpoints in `spec` are synced to the `status` section, allowing cluster operators to continue to use the `status` section and provide a place to perform validations.
+
+2. There is no great way for the user to track if the change is propagated to the all the cluster operators.
+
+##### Alternatives
+
+1. Create another global configuration for Cloud API endpoints that stores information like the endpoints themselves, trusted bundles etc.
+Infrastructure global configuration already performs the function of tracking infrastructure related configuration and another global configuration that stores a part of the information doesn't seem like a great option. But it might allow validations and status observation by an independent controller.
+
+2. Configure each individual cluster operator
+There are five cluster operators that would need to be configured namely, cluster-kube-controller-manager, cluster-ingress-operator, cluster-machine-api-operator, cluster-image-registry-operator, cluster-credential-operator. There might be more operators like cluster-network-operator that might require access to the AWS APIs in the future to control the security group rules. Configuring all these separately is not a great UX for installer and a user who wants to modify the cluster to use API endpoints as day-2 operation.
```

Comment:
I believe OLM operators like Metering (and any other operator leveraging AWS services) would need to likewise be configured.